filters/flowid/generator-ulid.go

```diff
@@ -0,0 +1,40 @@
+package flowid
+
+import (
+	"github.com/oklog/ulid"
+	"io"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+type ulidGenerator struct {
+	sync.Mutex
+	r io.Reader
+}
+
+func newULIDGenerator() flowIDGenerator {
+	return newULIDGeneratorWithEntropyProvider(rand.New(rand.NewSource(time.Now().UTC().UnixNano())))
+}
+
+func newULIDGeneratorWithEntropyProvider(r io.Reader) flowIDGenerator {
+	return &ulidGenerator{r: r}
+}
+
+func (g *ulidGenerator) Generate() (string, error) {
+	g.Lock()
```

Comment:
Creating new ones on every call will hurt the garbage collector. My first attempt resulted in:

```
BenchmarkFlowIdULIDGenerator/Std-8  	  100000	     11762 ns/op
```
For doing it like this instead:
```
	//g.Lock()
	r := rand.New(rand.NewSource(time.Now().UTC().UnixNano()))
	id, err := ulid.New(ulid.Now(), r)
	//g.Unlock()
```

I added details for all those attempts as comments for #251. A completely, non thread safe, usage of the same entropy provider wasn't faster than 170ns.